R/layers.R

```diff
@@ -237,12 +241,124 @@ addMarkers = function(
   options = markerOptions(),
   data = getMapData(map)
 ) {
-  options$icon = icon
+  options$icon = L.icon(evalFormula(icon, data))
+  options$iconGroup = attr(options$icon, 'iconGroup', exact = TRUE)
   pts = derivePoints(data, lng, lat, missing(lng), missing(lat), "addMarkers")
   invokeMethod(map, data, 'addMarkers', pts$lat, pts$lng, layerId, options, popup) %>%
     expandLimits(pts$lat, pts$lng)
 }
 
+L.icon = function(options, embed = TRUE) {
+  if (!is.list(options)) return(options)
+  if (is.null(names(options))) {
+    if (length(options) == 0) return()
+    # in theory we need to check all sub-lists and all must be named
+    if (is.null(names(options[[1]])))
+      stop('The data for an individual icon must be a named list')
+    # it is already a list of icons, so no further base64 encoding
+    return(options)
+  }
+  image_uri = getFromNamespace('.b64EncodeFile', 'markdown')
```

Comment:
Yes, base64enc requires users to provide the mime info. I did not want to write a similar wrapper function like `markdown:::.b64EncodeFile` to make use of `mime::guess_type()`. Since the private function is in our own hands, I'm not too worried about it. 

I think it is a good idea if you can just bring in Rcpp, provided that we are going to use C++ anyway in the future. Thanks!